src/Shared/test/CommandLineApplicationTests.cs

```diff
@@ -1166,5 +1166,28 @@ public void ThrowsExceptionOnInvalidOption()
 
             Assert.Equal($"Unrecognized option '{inputOption}'", exception.Message);
         }
+
+        [Fact]
+        public void TreatUnmatchedOptionsAsArguments()
```

Comment:
Can you:

1. Add an actual option to this test and validate that it **does** get properly matched.
2. Add a second test that is essentially the same but asserts the behavior when `treatUnmatchedOptionsAsArguments` is explicitly set to `false`? (It should fail because of the unmatched options).